test/server/admin/prometheus_stats_test.cc

```diff
@@ -98,67 +99,93 @@ class PrometheusStatsFormatterTest : public testing::Test {
 
   void clearStorage() {
     pool_.clear();
-    counters_.clear();
-    gauges_.clear();
-    histograms_.clear();
-    textReadouts_.clear();
+    test_scope_ptr_->counters_.clear();
```

Comment:
I think you can just do test_scope_ptr_.reset() instead of all this.

Reply:
Hm `numSymbols()` of the symbol table doesn't seem to clear though when using `test_scope_ptr_ .reset()`?

Reply:
Hm. That seems surprising to me. Could you try calling symbol_table_->debugPrint() to see what's still there?

Reply:
never mind I know what the problem is. For that implementation, the scopes don't really own the stats; the Store does.

You'd have to clear out the MockStore, which is annoying since it's not a shared_ptr. What you have is fine. Sorry for the confusion. You could leave a comment saying why we have to reach into the Store to clean all the stats in it, and maybe we could clean it up later.